### ISSUE TITLE
btrbk resolves wrong mountpoint when using autofs

### ISSUE THREAD

**Issue by Massimo-B** [digint/btrbk#247]
Almost the same setup is working on Gentoo Linux with OpenRC, but failing on OpenSuse LEAP 15.0:

Source package : btrfsprogs-4.15-lp150.1.2.src
4.12.14-lp150.12.13-default

Recent version from github:
```
# btrbk --version
btrbk command line client, version 0.27.0-dev
```

btrbk.conf:
```
archive_preserve        12m 10y
archive_preserve_min    latest
incremental             yes
lockfile                /var/log/btrbk.lock
stream_buffer           5%
timestamp_format        long-iso
transaction_log         /var/log/btrbk.log

volume /mnt/btrfs-top-lvl/
    snapshot_create  onchange

    
    subvolume @/home
        group home
        snapshot_dir            @/.snapshots/home
        snapshot_preserve_min   3d
        snapshot_preserve       4d 4w 2m 1y

        target send-receive   	/mnt/usb/mobiledata/snapshots/balintop/home
            target_preserve_min     3d
            target_preserve         6d 5w 4m 3y
```

From -l trace:
```
Initializing target section: /mnt/usb/mobiledata/snapshots/balintop/home
Resolving btrfs mount point for: /mnt/usb/mobiledata/snapshots/balintop/home
### readlink -e /mnt/usb/mobiledata/snapshots/balintop/home
... Executing command: readlink -e /mnt/usb/mobiledata/snapshots/balintop/home
... Command output:
/mnt/usb/mobiledata/snapshots/balintop/home
Command execution successful
Real path for "/mnt/usb/mobiledata/snapshots/balintop/home" is: /mnt/usb/mobiledata/snapshots/balintop/home
... mountpoint_cache HIT: localhost
... resolved mount point (spec=systemd-1, subvolid=<undef>): /mnt/usb/mobiledata
No btrfs mount point found for: /mnt/usb/mobiledata/snapshots/balintop/home
WARNING: Skipping target "/mnt/usb/mobiledata/snapshots/balintop/home": Failed to fetch subvolume detail
```

**Comment by digint** [comment]
Thanks for reporting, it's a bug on how btrbk 0.27.0 resolves mount points (which is a new feature allowing btrbk to identify identical btrfs file systems on different mounts).

From your bpaste:
```
# /proc/self/mounts
# [...]
systemd-1 /mnt/usb/mobiledata autofs rw,relatime,fd=23,pgrp=1,timeout=300,minproto=5,maxproto=5,direct,pipe_ino=15826 0 0
# [...]
/dev/mapper/mobiledata_crypt /mnt/usb/mobiledata btrfs rw,noatime,nodiratime,compress-force=zlib,nossd,space_cache,autodefrag,subvolid=5,subvol=/ 0 0
```

If autofs is running, `/proc/self/mounts` shows the mount point twice, once with type=autofs, and once with type=btrfs (if auto-mounted).

**Fixed in**: 37c0e840e9139adf239596d6a22073f87bae8a77

I also changed btrbk to read `/proc/self/mountinfo` instead of `/proc/self/mounts`, which lacks extra info (namespaces) and is not documented in kernel (deprecated?): e02c2cf24973ec7991c8b13b06797ca1e19754dd, 069304f3bbc6f8573d1551bdb3999b52e1577055

**Comment by Massimo-B** [comment]
Working, thanks.
